File: CPP/program/Parking.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include <vector>
#include "Motorcycle.cpp"
#include "Car.cpp"


// Deklarasi namespace
using namespace std;


class Parking {
private:
    string nama;
    int kapasitas;
    int luas;
    int jumlahKendaraanSaatIni;
    vector<Motorcycle> listMotor;
    vector<Car> listMobil;

public:
    // Constructor
    Parking()
    {
    }
    
    // Constructor dengan parameter lengkap
    Parking(string nama, int kapasitas, int luas, int jumlahKendaraanSaatIni) {
        this->nama = nama;
        this->kapasitas = kapasitas;
        this->luas = luas;
        this->jumlahKendaraanSaatIni = jumlahKendaraanSaatIni;
        this->listMotor = {};
        this->listMobil = {};
    }
    
    Parking(string nama, int kapasitas, int luas, int jumlahKendaraanSaatIni, vector<Car> mobil) {
        this->nama = nama;
        this->kapasitas = kapasitas;
        this->luas = luas;
        this->jumlahKendaraanSaatIni = jumlahKendaraanSaatIni;
        this->listMotor = {};
        listMobil = mobil;
    }
    Parking(string nama, int kapasitas, int luas, int jumlahKendaraanSaatIni, vector<Motorcycle> motor) {
        this->nama = nama;
        this->kapasitas = kapasitas;
        this->luas = luas;
        this->jumlahKendaraanSaatIni = jumlahKendaraanSaatIni;
        listMotor = motor;
        this->listMobil = {};
    }
    Parking(string nama, int kapasitas, int luas, int jumlahKendaraanSaatIni, vector<Car> mobil, vector<Motorcycle> motor) {
        this->nama = nama;
        this->kapasitas = kapasitas;
        this->luas = luas;
        this->jumlahKendaraanSaatIni = jumlahKendaraanSaatIni;
        listMotor = motor;
        listMobil = mobil;
    }


    // Getter untuk mengambil nilai atribut nama
    string getNama() {
        return nama;
    }

    // Setter untuk mengatur nilai atribut nama
    void setNama(string nama) {
        this->nama = nama;
    }

    // Getter untuk mengambil nilai atribut kapasitas
    int getKapasitas() {
        return kapasitas;
    }

    // Setter untuk mengatur nilai atribut kapasitas
    void setKapasitas(int kapasitas) {
        this->kapasitas = kapasitas;
    }

    // Getter untuk mengambil nilai atribut luas
    int getLuas() {
        return luas;
    }

    // Setter untuk mengatur nilai atribut luas
    void setLuas(int luas) {
        this->luas = luas;
    }

    // Getter untuk mengambil nilai atribut jumlahKendaraanSaatIni
    int getJumlahKendaraanSaatIni() {
        return jumlahKendaraanSaatIni;
    }

    // Setter untuk mengatur nilai atribut jumlahKendaraanSaatIni
    void setJumlahKendaraanSaatIni(int jumlahKendaraanSaatIni) {
        this->jumlahKendaraanSaatIni = jumlahKendaraanSaatIni;
    }

    void setListMotor(vector<Motorcycle> motor)
    {
        listMotor = motor;
    }

    // Getter untuk vektor Motor
    vector<Motorcycle> getListMotor() { 
        return listMotor; 
    }

    void setListMobil(vector<Car> mobil)
    {
        listMobil = mobil;
    }
    
    // Getter untuk vektor Mobil
    vector<Car> getListMobil() { 
        return listMobil; 
    }

    // Destructor
    ~Parking()
    {
    }
};

File: Python/program/Parking.py
from Car import Car
from Motorcycle import Motorcycle
# Kelas Parking
class Parking :
    __nama =""
    __kapasitas=0
    __luas=0
    __jumlahKendaraanSaatIni=0

    # Constructor
    def __init__(self, nama, kapasitas=0, luas=0, jumlahKendaraanSaatIni=0, listMobil = [], listMotor=[]) :
        self.__nama = nama
        self.__kapasitas = kapasitas
        self.__luas = luas
        self.__jumlahKendaraanSaatIni = jumlahKendaraanSaatIni
        self.__listMobil = listMobil
        self.__listMotor = listMotor

    # Getter untuk mengambil nilai atribut nama
    def getNama(self) :
        return self.__nama

    # Setter untuk mengatur nilai atribut nama
    def setNama(self, nama) :
        self.__nama = nama

    # Getter untuk mengambil nilai atribut kapasitas
    def getKapasitas(self) :
        return self.__kapasitas

    # Setter untuk mengatur nilai atribut kapasitas
    def setKapasitas(self, kapasitas) :
        self.__kapasitas = kapasitas

    # Getter untuk mengambil nilai atribut luas
    def getLuas(self) :
        return self.__luas
    
    # Setter untuk mengatur nilai atribut luas
    def setLuas(self, luas) :
        self.__luas = luas
    
    # Getter untuk mengambil nilai atribut jumlahKendaraanSaatIni
    def getJumlahKendaraanSaatIni(self) :
        return self.__jumlahKendaraanSaatIni

    # Setter untuk mengatur nilai atribut jumlahKendaraanSaatIni
    def setJumlahKendaraanSaatIni(self, jumlahKendaraanSaatIni) :
        self.__jumlahKendaraanSaatIni = jumlahKendaraanSaatIni

    # Setter untuk listmotor
    def setListMotor(self, motor):
        listMotor = motor

    # Getter untuk listMotor
    def getListMotor(self) : 
        return self.__listMotor 

    # Setter untuk listmobil    
    def setListMobil(self, mobil):
        listMobil = mobil
    
    # Getter untuk listMobil
    def getListMobil(self) : 
        return self.__listMobil 